sky/adaptors/aws.py

```diff
@@ -93,58 +96,52 @@ def session():
 
 
 @import_package
+@functools.lru_cache()
```

Comment:
See the other comment. I feel we should
- revert to thread local (much easier to not be wrong)
- to limit the size of the cache, we can remove `**kwargs` here and support (based on looking at how resource() is used): `region_name`, `max_attempts`, `**aws_credentials` (this seems unused and can even be removed). This limits the surface a lot.
  - if we really want to limit the size of the cache, can build a `LRU` based on a thread local OrderedDict a la https://docs.python.org/3.7/library/collections.html#ordereddict-examples-and-recipes

Reply:
Please refer to the comment above. I am not sure why thread local will work better than `lru_cache` for readability. : )